src/main/java/net/fabricmc/loader/impl/transformer/ClassStripper.java

```diff
@@ -58,13 +62,41 @@ public void visit(int version, int access, String name, String signature, String
 
 	@Override
 	public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
-		if (stripFields.contains(name + descriptor)) return null;
-		return super.visitField(access, name, descriptor, signature, value);
+		if (stripFields.contains(name + descriptor)) {
+			return null;
+		} else {
+			return super.visitField(access, name, descriptor, signature, value);
+		}
 	}
 
 	@Override
 	public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
-		if (stripMethods.contains(name + descriptor)) return null;
-		return super.visitMethod(access, name, descriptor, signature, exceptions);
+		if (stripMethods.contains(name + descriptor)) {
```

Comment:
More of a problem with the existing code than your changes but using fresh strings as hashtable keys is very slow because each one needs a fresh hashcode computation. Should be a record instead.

Reply:
Sounds overkill, allocations are more or less free and this is especially elidable by the JIT. Point is allocating a class which holds both strings is far far cheaper than concatenation and a subsequent uncached hashCode computation.

Reply:
Method descriptors can be very long. Regardless I'm not saying it's necessarily a real performance issue, just a code smell IMO